### PR TITLE
Add event API

### LINKED ISSUE #4714: Implement log appender / event api
The spec is going to soon have a [log appender / event API](https://github.com/open-telemetry/opentelemetry-specification/pull/2676), in addition to the existing SDK.

To implement we need:
- [x] Rename classes for alignment. We have `LogEmitterProvider` and `LogEmitter` right now, we're going to need `LoggerProvider` and `Logger`. The rename is going to sting quite a bit because stable artifacts like `OpenTelemetrySdk` have references to the old names. We justified this because although `OpenTelemetrySdk` is stable, you need to take a dependency on the log SDK in order to use the log portions of the API. We'll need to make one time exception rules for jApiCmp to pass. #4721
- [x] Break out `:api:logs` module, including interfaces like `LoggerProvider`, `Logger`, `LogRecordBuilder`, etc. #4764
- [x] Add `GlobalLoggerProvider` akin to the old `GlobalMeterProvider`, which will hold global reference while until log stability is reached and it can merge with `GlobalOpenTelemetry`. #4764
- [ ] Extend API for events. `Logger` should have a `logRecordBuilder()` for use by log appenders, and `eventBuilder(String name)` for events. 
- [x] Rename `LogProcessor`, `*LogProcessor` to `LogRecordProcessor`, `*LogRecordProcessor`. #4772 
- [x] Rename `LogExporter`, `*LogExporter` to `LogRecordExporter`, `*LogRecordExporter`. #4773 
- [x] Rename `LogData` to `LogRecordData`. #4774

When this is complete the instrumentation repo will be able to delete [instrumentation-appender-api-internal](https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/instrumentation-appender-api-internal) and [instrumentation-appender-sdk-internal](https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/instrumentation-appender-sdk-internal) 🎉.

### PR DESCRIPTION
**Pull request by jack-berg**
Resolves #4714.

### CODE REVIEW

File: api/logs/src/main/java/io/opentelemetry/api/logs/DefaultLogger.java [hunk 1]
```diff
@@ -12,22 +12,36 @@
...
+  static Logger getInstance(boolean hasDomain) {
+    return hasDomain ? INSTANCE_WITH_DOMAIN : INSTANCE_NO_DOMAIN;
+  }
+
+  @Override
+  public EventBuilder eventBuilder(String name) {
+    if (!hasDomain) {
+      throw new IllegalStateException(
+          "Cannot emit event from Logger without event domain. Please use LoggerBuilder#setEventDomain(String) when obtaining Logger.");
+    }
```
Comment (trask) [inline api/logs/src/main/java/io/opentelemetry/api/logs/DefaultLogger.java:35]: is it allowed to throw exception in API? (vs log error and return noop)

Comment (jack-berg) [reply api/logs/src/main/java/io/opentelemetry/api/logs/DefaultLogger.java:35]: Logging and returning noop is more consistent with the other API level checks we have. 👍 

Will update.

**Review by jkwatson — APPROVED** [review]
I'm not 100% convinced that this API will end up being particularly user-friendly, but I'd like to get it into the project and try it out and see how it feels when used "in anger" before I pass final judgement on it.

**Comment by jack-berg** [comment]
> I'm not 100% convinced that this API will end up being particularly user-friendly

We'll see how it goes! I'm not particularly fond of having to use a bunch of `Logger*` classes to emit events. Maybe we can alleviate confusion through good documentation, maybe not. I hope this prototype produces the type of feedback we need to improve our comfort with this design, or change course.

**Pull request merged**